app/src/main/java/com/iitdh/sonusourav/instigo/Utils/CommonFunctions.java

```diff
@@ -83,13 +91,34 @@ public static boolean navigationItemSelect(MenuItem item, final Activity activit
                 PreferenceManager preferenceManager=new PreferenceManager(activity.getApplicationContext());
                 preferenceManager.setIsLoggedIn(false);
                 preferenceManager.setLoginCredentials("email","password");
-                FirebaseAuth loginAuth=FirebaseAuth.getInstance();
-                loginAuth.signOut();
 
-                Toast.makeText(activity.getApplicationContext(), "User successfully logged out.", Toast.LENGTH_SHORT).show();
-                activity.startActivity(new Intent().setClass(activity, LoginActivity.class));
+                FirebaseAuth loginAuth=FirebaseAuth.getInstance();
+                GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                        .requestIdToken(activity.getResources().getString(R.string.default_web_client_id))
+                        .requestEmail()
+                        .build();
+                GoogleSignInClient googleSignInClient = GoogleSignIn.getClient(activity, gso);
+
```

Comment:
If part is always equal to false, you can remove that. And put loginAuth.signOut() just after you defined Firebase Auth

Reply:
googleSignInClient will be null when the user is signed in with email, that block is intended to sign out such users and the else block is intended to sign out users who login through Goggle Sign In

Reply:
Cool, and why isn't loginAuth.signOut() in else part also?

Reply:
loginAuth.signOut() is needed for signing out from email. For google sign in, googleSignInClient is sufficient.

Reply:
Are you sure? Checkout this https://stackoverflow.com/a/38707954/8616229

Reply:
Sorry! My mistake. I was reading the Google sign in docs instead of the Firebase docs.